### ISSUE TITLE
False positive no-member with property on metaclass

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#378]
Originally reported by: **BitBucket: [habilain](http://bitbucket.org/habilain), GitHub: @habilain?**

---

The preferred method to use property for a Class variable is to place the property methods on the metaclass as follows:

```
#!python
class Meta(type):
  @property
  def test(cls):
    return 'Test'

class Class(metaclass=Meta):
  pass

print(Class.test)
```

However, the above code when run through PyLint generates a false no-member message for Class.test (Note that while I supplied Python3 code, this also occurs for Python2).

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/377

**Comment by PCManticore** [comment]
Closing as will most likely not support `setattr` in the foreseeable future.
